Zone.UmbracoPersonalisationGroups/Helpers/EmbeddedResourceHelper.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace Zone.UmbracoPersonalisationGroups.Helpers
+{
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+
+    using Umbraco.Core;
+
+    using Zone.UmbracoPersonalisationGroups.Controllers;
+
+    using Constants = Zone.UmbracoPersonalisationGroups.AppConstants;
+
+    /// <summary>
+    /// Provides methods for retrieving embedded resources.
+    /// </summary>
+    internal class EmbeddedResourceHelper
+    {
+        /// <summary>
+        /// Returns a value indicating whether the given resource exists.
+        /// </summary>
+        /// <param name="resource">The resource name.</param>
+        /// <returns>
+        /// The <see cref="bool"/>.
+        /// </returns>
+        public static bool ResourceExists(string resource)
+        {
+            // Sanitize the resource request.
+            string root = Constants.ResourceRoot;
+            string criteria = Constants.ResourceForCriteriaRoot;
+            string extension = Constants.ResourceExtension;
+
+            if (resource.StartsWith(root))
+            {
+                resource = resource.TrimStart(root).Replace("/", ".").TrimEnd(extension);
+            }
+            else if (resource.StartsWith(criteria))
+            {
+                resource = resource.TrimStart(criteria).Replace("/", ".").TrimEnd(extension);
+            }
+            else if (resource.EndsWith(extension))
+            {
+                resource = resource.TrimEnd(extension);
+            }
+
+            // Get this assembly.
+            Assembly assembly = typeof(ResourceController).Assembly;
```

Comment:
I think there's possibly a slight issue here - criteria *could* by defined in another assembly, possibly a solution project, or maybe even a "contrib" type package in future.  You'll see above in GetResourceForCriteria() I'm using `criteria.GetType().Assembly` to look for resources within.  So probably the assembly needs passing to this helper method.